contracts/docs/payload-types.md

```diff
@@ -239,7 +244,15 @@ def update_rewards(
         else:
             liquidity -= tick.net_liquidity
 
+        reward_checksum = keccak_256(abi_encode_packed(
+            (reward_checksum, 'bytes32'),
+            (liquidity, 'uint128'),
+            (tick, 'int24')
+        ))
```

Comment:
looks like this example is incorrect because you overwrite reward_checksum every iter so it becomes only a checksum over the last tick.

Reply:
This is purposefully done so that it builds a hash chain `h(h(h(h(..., t1), t2), t3), t4)`

Reply:
ah yeah i missed the fact that you re-hash reward_checksum